src/interfaces/index.ts

```diff
@@ -2,11 +2,11 @@ export interface IPaginationOptions {
   /**
    * the amount of items to be requested per page
    */
-  limit: number;
+  limit: number | string;
```

Comment:
Probably best not to imply that the limit or page can be a string? What do you think?

Reply:
I guess this was for the tests after checking them out? Might as well use a ts-ignore just for the tests. I would do this as this is the scenario we're testing against right? Because this is a JS issue and not a TS one.

Reply:
Yeah at first I was also thinking that this should remain as strictly a number. But then my thinking ended up being something like: "we support string values for these parameters so is there a good reason not to type it as such?".

The reason we have the line

```
const resolvedValue = Number(value);
```

is because `value` can be a string. So I think it makes sense to type it as such.

Anyway I don't feel strongly so let me know if you'd like me to change it to just `number` and then I'll also update the tests to pass `any` (or use `ts-ignore`) like you suggest. :)

Reply:
Yea I see your point however a string is not always `"12"` you know? A string is also `"hello"` so I wanted to avoid string values. I guess this scenario only exists because of either unstrict types or type casting from the param decorator where `(page: number)` or `(page)` could result in page being a string when running in JS. Let's ask a third person just to be sure, @zMotivat0r what do you think?

Reply:
I see your point too :)

I guess one other thing to point out is we already have the same issue with `number`. A number is not always `12` - a number is also `12.34`.

Ideally our allowed type would be something like `PositiveInteger | StringPositiveInteger` but possibly `number | string` is as close as we can reasonably get?

Reply:
In fact based off the above I've just pushed a new commit to make negative or decimal values for `page` and `limit` fallback to the default. https://github.com/nestjsx/nestjs-typeorm-paginate/pull/427/commits/1cd0732703b0491dbcd881a92672ca5b85a5f655